extension/packagemanagerextension/nbextension/tree.js

```diff
@@ -0,0 +1,68 @@
+define(function (require) {
+    var $ = require('jquery');
+    var Jupyter = require('base/js/namespace');
+    var utils = require('base/js/utils');
+    var scripts = require('./scripts');
+    var urls = require('./urls');
+    var methods = require('./methods');
+
+    function init(dir) {
+        scripts.packageview.load(dir);
+        scripts.searchview.load();
+        scripts.closeview.load();
+        methods.updatepkg.load();
+        methods.deletepkg.load();
+        methods.installpkg.load();
+    };
+
+
+    function load() {
```

Comment:
The load basically should do nothing in our extension, since the call to open the panel is done somewhere else (contents manager, for example, when the user clicks the cog button).

Reply:
You should also expose another function that checks in the backend if there's any package not installed (or some other action the users needs to do). I want the cog button to display a warning sign on top if that is the case (check the document I shared).

Reply:
Is it not implemented in the package_info API itself? I'm sending the status of every package along with the payload. I'm currently using that to check if the packages are in sync (either of the 3 cases we discussed). For the cog button, I need to clarify on when I would have to do the API call to check the status. That is specific to SWAN and not Vanilla Jupyter, right?

Reply:
Yes, the information comes from the API and you just need to expose a function that will give the status of the project ("everything fine" or "attention needed"), by checking the sync status.
This API call should happen every time the function I'm asking you to implement is called. As simple as that. When this is called, will be implemented in the contents manager (where you are defining the logic to show/hide the cog button).
But then we need to optimise, probably.